classes/Combination.php

```diff
@@ -276,6 +279,16 @@ public function deleteAssociations()
         return $result;
     }
 
+    /**
+     * Delete product combination from cart.
+     *
+     * @return bool
+     */
+    public function deleteCartProductCombination(): bool
```

Comment:
```suggestion
    protected function deleteCartProductCombination(): bool
```

I see no reason to have this method public.
Also, you should test if `$this->id` is correctly defined before running an SQL query for nothing.